tiledb/sm/filesystem/gcs.cc

```diff
@@ -127,20 +130,83 @@ Status GCS::init(const Config& config, ThreadPool* const thread_pool) {
   return Status::Ok();
 }
 
+/**
+ * Builds a chain of service account impersonation credentials.
+ *
+ * @param credentials The set of credentials to start the chain.
+ * @param service_accounts A comma-separated list of service accounts, where
+ * each account will be used to impersonate the next.
+ * @options Options to set to the credentials.
+ * @return The new set of credentials.
+ */
+static shared_ptr<google::cloud::Credentials> apply_impersonation(
+    shared_ptr<google::cloud::Credentials> credentials,
+    std::string service_accounts,
+    google::cloud::Options options) {
+  if (service_accounts.empty()) {
+    return credentials;
+  }
+  auto last_comma_pos = service_accounts.rfind(',');
```

Comment:
What are the requirements for service account names? Please make sure we are handling cases like space in the name if that's allowed, or else assert that we error.

Reply:
If the service account name is invalid the GCS SDK will be the one that will fail. I don't think it is a good idea to duplicate the validation logic in our own code.

Reply:
As far as I understood when I reviewed this initially there weren't any requirements drafted, that's why I didn't approve, IIRC passing `,` breaks the current code.

Reply:
If the impersonation string is `","` the account to impersonate is `""` and there is one delegate `""`. I ran it and will commit an additional test case, there is no sefgault if that's what you mean.

Reply:
just as a reference, Google IAM service account names look like <code><var>service-account-name</var>@<var>project-name</var>.iam.gserviceaccount.com</code>; i.e. they look like email addresses (and it may? be possible to impersonate a user account as well?). at any rate I think letting the Google client library handle validation is the right way to do it.

Reply:
Added test for `","`.

Reply:
Thanks